haskell/private/actions/info.bzl

```
@@ -0,0 +1,174 @@
+            # TODO: currently, this will duplicate the common, target-independent options for
+            # each build target.  We should instead move them into GhcConfig.common_options.
+            options = _filter_package_env(c.compile_flags),
+            transitive_cc_libs = [lib.path for lib in cc_libs],
+            # Follows the new runfiles tree organization of:
+            # https://github.com/bazelbuild/bazel/wiki/Updating-the-runfiles-tree-structure
+            runfiles = [
+                struct(
+                    full_path = f.path,
+                    short_path = paths.join(
+                        f.owner.workspace_name or workspace_name, f.short_path))
+                for f in runfiles.to_list()
+            ],
+        ),
+    )
+
+
+def library_info_output_groups(
+        name,
+        hs,
+        hs_info,
+        lib_info):
+    """Output groups for depending on a Haskell target.
+
+    Args:
+        name: A string; the name of the current target.
+        hs: The Haskell context.
+        hs_info: A HaskellInfo provider.
+        lib_info: A HaskellLibraryInfo provider.
+
+    Returns:
+      A dict whose keys are output groups and values are depsets of Files.
+    """
+    proto_file = write_proto_file(
+        hs = hs,
+        output_name = name + ".HaskellLibrary",
+        proto_type = "haskell.LibraryInfo",
+        content = struct(
+            # TODO(google/hrepl#4): currently, we only expose the immediate dependencies.
+            transitive_package_ids = [lib_info.package_id],
+            transitive_package_dbs =
+                [db.dirname for db in hs_info.package_databases.to_list()]))
+    return {
+        "haskell_transitive_deps": depset(
+          transitive = [
+                hs_info.package_databases,
+                hs_info.interface_dirs,
+                hs_info.dynamic_libraries,
+            ]),
+        "haskell_library_info": depset([proto_file]),
+    }
+
+def compile_info_output_groups(
+        name,
+        workspace_name,
+        hs,
+        c,
+        posix,
+        cc_info,
+        runfiles):
+    """Output groups for compiling a Haskell target.
+
+    Args:
+        name: A string; the name of the current target.
+        workspace_name: The workspace this target was defined in.
+          Used for organizing its runfiles.
+        hs: The Haskell context.
+        c: A struct with information about the compilation step.
+        posix: The posix toolchain.
+        cc_info: A CcInfo provider.
+        runfiles: A depset of Files.
+
+    Returns:
+      A dict whose keys are output groups and values are depsets of Files.
+    """
+    (ghci_extra_libs, ghc_env) = get_ghci_extra_libs(hs, posix, cc_info)
+    cc_libs = [lib for lib in ghci_extra_libs.to_list()
+                if not is_hs_library(get_lib_name(lib)) and get_lib_name(lib) != "ffi"]
+    return {
+        "haskell_cdeps_shared_lib": depset(cc_libs),
```

```suggestion
        "haskell_cdeps_lib": depset(cc_libs),
```
Since `cc_libs` doesn't distinguish static and shared libraries. I.e. same reasoning as [here](https://github.com/tweag/rules_haskell/pull/1210/files#r367520407).

---

Done. (Renamed to `haskell_cdep_libs` to be a little more like the other providers.)

---

Thanks!